### PR TITLE
Improve workflow

### PR DESCRIPTION
**Pull request by szepeviktor**

### CODE REVIEW

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -3,23 +3,26 @@ name: E2E test
...
   cypress:
-    name: ${{ matrix.core.name }}
+    name: ${{ matrix.job_name }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        core:
-          - {name: 'WP latest', version: 'latest'}
-          - {name: 'WP minimum', version: 'WordPress/WordPress#5.7'}
-          - {name: 'WP trunk', version: 'WordPress/WordPress#master'}
```
Comment (szepeviktor) [inline .github/workflows/cypress.yml:19]: Objects are invalid matrix elements.

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -3,23 +3,26 @@ name: E2E test
...
+        - job_name: 'WP latest'
+          core_version: 'latest'
+        - job_name: 'WP minimum'
+          core_version: 'WordPress/WordPress#5.7'
+        - job_name: 'WP trunk'
+          core_version: 'WordPress/WordPress#master'
     steps:
     - name: Checkout
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
```
Comment (szepeviktor) [inline .github/workflows/cypress.yml:25]: Upgrade this action

File: .github/workflows/cypress.yml (hunk 1 above)
Comment (psorensen) [reply .github/workflows/cypress.yml:19]: If this is removed, how will GH Actions know which versions to test against? This appears to be working fine: https://github.com/psorensen/curated-query-loop/actions/runs/7579645133

Comment (szepeviktor) [reply .github/workflows/cypress.yml:19]: Yes it appears to work but not supported.
A proper matrix is added in the next lines.

Comment (psorensen) [reply .github/workflows/cypress.yml]: Oops! I missed this. Thanks!

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**